keepvariable/keepvariable_core.py

```diff
@@ -513,36 +552,47 @@ def json_mset(self, name: str, params: dict[str, Any], transaction=True) -> None
             pipe.execute()
 
     def query(
-        self, query_params: dict, index_name: str, *args,
+        self, *, text_params: Optional[dict[str, tuple]] = None,
+        tag_params: Optional[dict[str, tuple]] = None, index_name: str,
         field_to_sort_by: Optional[str] = None, asc=True, **kwargs
-    ) -> list[str]:
+    ) -> list[tuple]:
         """
-        Simplified wrapper to RedisSearch. Allows to search and sort by a value of Redis TAG fields.
+        Simplified wrapper to RedisSearch. Allows to search and sort by a value of Redis TAG/TEXT fields.
         Simplistic on purpose, to avoid bloat. Additional functionality should be added in case of need.
 
-        :param query_params: attribute name to value mapping, e.g. {'status': 'QUEUED', ...}
-        :type query_params: dict
+        :param text_params: key name to a tuple of values mapping, e.g. {'status': ('pipel', ...), ...}
+        :type text_params: dict[str, tuple]
+        :param tag_params: key name to a tuple of values mapping, e.g. {'status': ('QUEUED', ...), ...}
+        :type tag_params: dict[str, tuple]
         :param index_name: name of the Redis index used for querying
         :type index_name: str
         :param field_to_sort_by: attribute name by which results should be sorted, defaults to None
         :type field_to_sort_by: Optional[str], optional
         :param asc: True if sort in ascending order, defaults to True
         :type asc: bool, optional
-        :return: list of found document jsons
+        :return: [('jobs:43', job_json), ...]
         :rtype: list[str]
         """
-        tag_query_template = "@{field}:{{{value}}}"
         final_query = ""
 
-        for field, value in query_params.items():
-            final_query += tag_query_template.format(field=field, value=value)
+        if text_params is not None:
+            text_query_template = "{field}:{value}"
+            for field, values in text_params.items():
+                value_str = "|".join(values)
+                final_query += text_query_template.format(field=field, value=value_str)
+
+        if tag_params is not None:
+            tag_query_template = "@{field}:{{{value}}}"
+            for field, values in tag_params.items():
+                value_str = "|".join(values)
+                final_query += tag_query_template.format(field=field, value=value_str)
```

Comment:
please add an example in comment how the final query can look like + how it was transformed